### ISSUE TITLE
1.19.2 worldgen lock on load

### ISSUE THREAD

**Issue by Zelioth** [J-Dill/NoMoWanderer#18]
Minecraft 1.19.2
NMW - 1.3.12
Forge - 43.3.5

"Server thread" #222 prio=4 os_prio=-1 cpu=21312.50ms elapsed=5366.29s tid=0x00000262567c84d0 nid=0x3170 runnable  [0x000000ae4f0fd000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@17.0.8/Native Method)
        - parking to wait for  <0x000000063b232d70> (a java.lang.String)
        at java.util.concurrent.locks.LockSupport.parkNanos(java.base@17.0.8/LockSupport.java:252)
        at net.minecraft.util.thread.BlockableEventLoop.m_5667_(minecraft@1.19.2/BlockableEventLoop.java:152)
        at net.minecraft.util.thread.BlockableEventLoop.m_18701_(minecraft@1.19.2/BlockableEventLoop.java:142)
        at net.minecraft.server.level.ServerChunkCache.getChunkBlocking(minecraft@1.19.2/ServerChunkCache.java:1762)
        at net.minecraft.server.level.ServerChunkCache.m_7587_(minecraft@1.19.2/ServerChunkCache.java:1658)
        at net.minecraft.world.level.Level.m_6522_(minecraft@1.19.2/MixinLevel.java:179)
        at net.minecraft.world.level.Level.m_46865_(minecraft@1.19.2/MixinLevel.java:4039)
        at net.minecraft.world.level.Level.m_46745_(minecraft@1.19.2/MixinLevel.java:4034)
        at nomowanderer.world.EntitySpawnHandler.getChunk(nomowanderer@1.19.2_1.3.12/EntitySpawnHandler.java:202)
        at nomowanderer.world.EntitySpawnHandler.checkSpawn(nomowanderer@1.19.2_1.3.12/EntitySpawnHandler.java:70)
        at nomowanderer.world.EntitySpawnHandler.maybeChangeEntitySpawn(nomowanderer@1.19.2_1.3.12/EntitySpawnHandler.java:41)
        at nomowanderer.world.__EntitySpawnHandler_maybeChangeEntitySpawn_EntityJoinLevelEvent.invoke(nomowanderer@1.19.2_1.3.12/.dynamic)
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(net.minecraftforge.eventbus/ASMEventHandler.java:73)
        at net.minecraftforge.eventbus.EventBus$$Lambda$5034/0x0000000800f5cce0.invoke(net.minecraftforge.eventbus/Unknown Source)
        at net.minecraftforge.eventbus.EventBus.post(net.minecraftforge.eventbus/EventBus.java:315)
        at net.minecraftforge.eventbus.EventBus.post(net.minecraftforge.eventbus/EventBus.java:296)
        at net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157538_(minecraft@1.19.2/PersistentEntitySectionManager.java:79)
        at net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157604_(minecraft@1.19.2/PersistentEntitySectionManager.java:121)
        at net.minecraft.world.level.entity.PersistentEntitySectionManager$$Lambda$80412/0x00000008082fb638.accept(minecraft@1.19.2/Unknown Source)
        at net.minecraft.world.entity.EntityType$1.m_147060_(minecraft@1.19.2/EntityType.java:529)
        at net.minecraft.world.entity.EntityType$1$$Lambda$80414/0x00000008082fba98.apply(minecraft@1.19.2/Unknown Source)
        at java.util.Optional.map(java.base@17.0.8/Optional.java:260)
        at net.minecraft.world.entity.EntityType.m_20645_(minecraft@1.19.2/EntityType.java:507)
        at net.minecraft.world.entity.EntityType$1.m_147056_(minecraft@1.19.2/EntityType.java:528)
        at net.minecraft.world.entity.EntityType$1$$Lambda$80413/0x00000008082fb868.accept(minecraft@1.19.2/Unknown Source)
        at java.util.ArrayList$ArrayListSpliterator.tryAdvance(java.base@17.0.8/ArrayList.java:1602)
        at net.minecraft.world.entity.EntityType$1.tryAdvance(minecraft@1.19.2/EntityType.java:527)
        at java.util.Spliterator.forEachRemaining(java.base@17.0.8/Spliterator.java:332)
        at java.util.stream.ReferencePipeline$Head.forEach(java.base@17.0.8/ReferencePipeline.java:762)
        at net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157559_(minecraft@1.19.2/PersistentEntitySectionManager.java:120)
        at net.minecraft.server.level.ServerLevel.m_143327_(minecraft@1.19.2/ServerLevel.java:1517)
        at net.minecraft.server.level.ChunkMap.m_143064_(minecraft@1.19.2/ChunkMap.java:666)
        at net.minecraft.server.level.ChunkMap.m_214898_(minecraft@1.19.2/ChunkMap.java:683)
        at net.minecraft.server.level.ChunkMap$$Lambda$80351/0x00000008082ea2c8.m_196866_(minecraft@1.19.2/Unknown Source)
        at net.minecraft.world.level.chunk.LevelChunk.m_62952_(minecraft@1.19.2/LevelChunk.java:429)
        at net.minecraft.server.level.ChunkMap.mixinextras$bridge$m_62952_$137(minecraft@1.19.2/ChunkMap.java)
        at net.minecraft.server.level.ChunkMap$$Lambda$80354/0x00000008082eae48.call(minecraft@1.19.2/Unknown Source)
        at net.minecraft.server.level.ChunkMap.wrapOperation$zgm000$setCurrentLoadingThenPostLoad(minecraft@1.19.2/ChunkMap.java:1854)
        at net.minecraft.server.level.ChunkMap.m_214854_(minecraft@1.19.2/ChunkMap.java:691)
        at net.minecraft.server.level.ChunkMap$$Lambda$80350/0x00000008082e9140.apply(minecraft@1.19.2/Unknown Source)
        at com.mojang.datafixers.util.Either.lambda$mapLeft$0(datafixerupper@5.0.28/Either.java:162)
        at com.mojang.datafixers.util.Either$$Lambda$80009/0x00000008082972d0.apply(datafixerupper@5.0.28/Unknown Source)
        at com.mojang.datafixers.util.Either$Left.map(datafixerupper@5.0.28/Either.java:38)
        at com.mojang.datafixers.util.Either.mapLeft(datafixerupper@5.0.28/Either.java:162)
        at net.minecraft.server.level.ChunkMap.m_214851_(minecraft@1.19.2/ChunkMap.java:675)
        at net.minecraft.server.level.ChunkMap$$Lambda$80347/0x00000008082e8ac0.apply(minecraft@1.19.2/Unknown Source)
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(java.base@17.0.8/CompletableFuture.java:646)
        at java.util.concurrent.CompletableFuture$Completion.run(java.base@17.0.8/CompletableFuture.java:482)
        at net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(minecraft@1.19.2/ChunkTaskPriorityQueueSorter.java:62)
        at net.minecraft.server.level.ChunkTaskPriorityQueueSorter$$Lambda$79841/0x000000080825d430.run(minecraft@1.19.2/Unknown Source)
        at net.minecraft.util.thread.BlockableEventLoop.m_6367_(minecraft@1.19.2/BlockableEventLoop.java:157)
        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_6367_(minecraft@1.19.2/ServerChunkCache.java:535)
        at net.minecraft.util.thread.BlockableEventLoop.m_7245_(minecraft@1.19.2/BlockableEventLoop.java:131)
        at net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_7245_(minecraft@1.19.2/ServerChunkCache.java:543)
        at net.minecraft.server.level.ServerChunkCache.m_8466_(minecraft@1.19.2/ServerChunkCache.java:267)
        at net.minecraft.server.MinecraftServer.m_129961_(minecraft@1.19.2/MinecraftServer.java:751)
        at net.minecraft.server.MinecraftServer.m_7245_(minecraft@1.19.2/MinecraftServer.java:740)
        at net.minecraft.util.thread.BlockableEventLoop.m_18701_(minecraft@1.19.2/BlockableEventLoop.java:140)
        at net.minecraft.server.MinecraftServer.m_130012_(minecraft@1.19.2/MinecraftServer.java:726)
        at net.minecraft.server.MinecraftServer.m_129940_(minecraft@1.19.2/MinecraftServer.java:447)
        at net.minecraft.server.MinecraftServer.m_130006_(minecraft@1.19.2/MinecraftServer.java:302)
        at net.minecraft.client.server.IntegratedServer.m_7038_(minecraft@1.19.2/IntegratedServer.java:62)
        at net.minecraft.server.MinecraftServer.m_130011_(minecraft@1.19.2/MinecraftServer.java:625)
        at net.minecraft.server.MinecraftServer.m_206580_(minecraft@1.19.2/MinecraftServer.java:244)
        at net.minecraft.server.MinecraftServer$$Lambda$78986/0x000000080812f2a0.run(minecraft@1.19.2/Unknown Source)
        at java.lang.Thread.run(java.base@17.0.8/Thread.java:833)

   Locked ownable synchronizers:
        - None

adding the mod and attempting to generate the world stops at 100% then doesn't go any further. Server thread for the load is doing this.

**Comment by J-Dill** [comment]
Thanks for the report! Were you using any other mods at the time that this happened?

**Comment by esvarc** [comment]
I also verified that the MOD causes the game to freeze when new chunks are generated. I originally thought it was MOD Town and Towers that were causing the freeze. In case I disabled the village_wandering_trader_camp generation the problematic chunk managed to load. However, I then tried to pre-generate chunks using MOD Chunky-1.3.92.jar and it froze in several different chunks. Then I started to first disable all possible MODs that optimize memory usage and generate new chunks, such as Canary, Memory Leak Fix, Smooth Boot, Smoothchunk, etc. However, none of that helped, but once I disabled NoMoWanderer everything was OK.

It is possible to download from onedrive (https://1drv.ms/u/s!AsuclbmnGMOphrNnYmqiRs6Egrjb6w?e=YbSPZ7) a ready-made instance including configuration and a world where the problem can be diagnosed. It is for MC V1.20.1 and forge 47.2.18. Before the test, just copy the pre-generated world ".\saves\Explore-Master"  into a new directory and do the test. Join the world and then give chat command

/teleport @a -640 ~ 144
or
/teleport @a -688 ~ 80

both chunks will cause freezing. After turning off NoMoWanderer, everything will generate correctly. The following error appears in the log

`[13:10:05] [ModernFix integrated server watchdog/ERROR] [or.em.mo.wo.IntegratedWatchdog/]: A single server tick has taken 40001, more than 40000 milliseconds`


```
"Server thread" prio=5 Id=144 TIMED_WAITING on java.lang.String@7d0fa43b
	at java.base@17.0.8.1/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.lang.String@7d0fa43b
---SNIP---
at TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.EntitySpawnHandler.getAABB(EntitySpawnHandler.java:127)
at TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.EntitySpawnHandler.canFindCancelEntity(EntitySpawnHandler.java:86)
at TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.EntitySpawnHandler.checkSpawn(EntitySpawnHandler.java:61)
at TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.EntitySpawnHandler.maybeChangeEntitySpawn(EntitySpawnHandler.java:39)
at TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.__EntitySpawnHandler_maybeChangeEntitySpawn_EntityJoinLevelEvent.invoke(.dynamic)
```


Mods:
  [✔️] v201-1.20.1-forge47.1.0-mod-pickupwidelysmp-v1.2.4.jar
  [✔️] v201-adlods-1.20.1-8.1.0.0-build.0142.jar
  [✔️] v201-advanced-xray-forge-1.20.1-2.18.1-build.22.jar
  [✔️] v201-ai-improvements-1.20-0.5.2.jar
  [✔️] v201-apotheosis-1.20.1-7.0.5.jar
  [✔️] v201-apothicattributes-1.20.1-1.1.4.jar
  [✔️] v201-appleskin-forge-mc1.20.1-2.5.1.jar
  [✔️] v201-appliedenergistics2-forge-15.0.15.jar
  [✔️] v201-architectury-9.1.12-forge.jar
  [✔️] v201-attributefix-forge-1.20.1-21.0.4.jar
  [✔️] v201-awesomedungeonend-forge-1.20.1-3.1.1.jar
  [✔️] v201-awesomedungeon-forge-1.20.1-3.2.0.jar
  [✔️] v201-awesomedungeonnether-forge-1.20.1-3.1.1.jar
  [✔️] v201-awesomedungeonocean-forge-1.20.1-3.3.0.jar
  [✔️] v201-backpacked-forge-1.20.1-2.2.5.jar
  [✔️] v201-balm-forge-1.20.1-7.2.1.jar
  [✔️] v201-betteradvancements-1.20.1-0.3.2.161.jar
  [✔️] v201-betterf3-7.0.2-forge-1.20.1.jar
  [✔️] v201-bettervillage-forge-1.20.1-3.2.0.jar
  [✔️] v201-biomesoplenty-1.20.1-18.0.0.598.jar
  [✔️] v201-block-swap-forge-1.20.1-5.0.0.0.jar
  [✔️] v201-bookshelf-forge-1.20.1-20.1.8.jar
  [✔️] v201-bountiful-6.0.3-1.20.1-forge.jar
  [✔️] v201-bygonenether-1.3.2-1.20.x.jar
  [✔️] v201-cagedmobs-1.20.1-forge-2.0.5.jar
  [✔️] v201-canary-mc1.20.1-0.2.7.jar
  [✔️] v201-carryon-forge-1.20.1-2.1.2.7.jar
  [✔️] v201-castle-dungeons-4.0.0-1.20-forge.jar
  [✔️] v201-catalogue-forge-1.20.1-1.8.0.jar
  [✔️] v201-cfm-forge-1.20.1-7.0.0-pre36.jar
  [✔️] v201-chisels-and-bits-forge-1.4.145.jar
  [✔️] v201-chunkloaders-1.2.8a-forge-mc1.20.1.jar
  [✔️] v201-chunksending-1.20.1-2.8.jar
  [✔️] v201-chunky-1.3.92.jar
  [✔️] v201-clienttweaks-forge-1.20-11.1.0.jar
  [✔️] v201-cloth-config-11.1.106-forge.jar
  [✔️] v201-clumps-forge-1.20.1-12.0.0.3.jar
  [✔️] v201-collective-1.20.1-7.22.jar
  [✔️] v201-colossalchests-1.20.1-1.8.4.jar
  [✔️] v201-controllable-forge-1.20.1-0.20.3.jar
  [✔️] v201-controlling-forge-1.20.1-12.0.2.jar
  [✔️] v201-crafttweaker-forge-1.20.1-14.0.27.jar
  [✔️] v201-cristellib-1.1.5-forge.jar
  [✔️] v201-ctov-3.3.6.jar
  [✔️] v201-cupboard-1.20.1-2.1.jar
  [✔️] v201-curios-forge-5.4.6-1.20.1.jar
  [✔️] v201-cyclic-1.20.1-1.12.4.jar
  [✔️] v201-cyclopscore-1.20.1-1.18.14.jar
  [✔️] v201-darkutilities-forge-1.20.1-17.0.3.jar
  [✔️] v201-deepercaves-1.20.1-1.2.3.jar
  [✔️] v201-deuf-1.20.1-1.3.jar
  [✔️] v201-dreadsteel-1.20.1-1.1.6.2.jar
  [✔️] v201-dungeon-crawl-1.20.1-2.3.14.jar
  [✔️] v201-dungeonsarise-1.20.1-2.1.57-release.jar
  [✔️] v201-easy-piglins-1.20.1-1.0.1.jar
  [✔️] v201-easy-villagers-1.20.1-1.0.17.jar
  [✔️] v201-elevatorid-1.20.1-1.8.15.jar
  [✔️] v201-embeddium-0.2.15-mc1.20.1.jar
  [✔️] v201-enchantinginfuser-v8.0.2-1.20.1-forge.jar
  [✔️] v201-enchantsmith-forge-1.0.3-1.20.jar
  [✔️] v201-enderite-cj-forge-mc1.20.1-1.4.2.jar
  [✔️] v201-everycomp-1.20-2.6.18.jar
  [✔️] v201-explorations-forge-1.20.1-1.5.2.jar
  [✔️] v201-explorify-v1.3.0-mc1.20u1.20.1.jar
  [✔️] v201-extraarmor-1.20.1-1.17.4.jar
  [✔️] v201-fastfurnace-1.20.1-8.0.1.jar
  [✔️] v201-fastload-reforged-mc1.20.1-3.4.0.jar
  [✔️] v201-fastsuite-1.20.1-5.0.1.jar
  [✔️] v201-fastworkbench-1.20.1-8.0.2.jar
  [✔️] v201-ferritecore-6.0.1-forge.jar
  [✔️] v201-flatbedrock-forge-3.0.1-build.18-mc1.20.1.jar
  [✔️] v201-flib-1.20.1-0.0.11.jar
  [✔️] v201-forgeconfigscreens-v8.0.2-1.20.1-forge.jar
  [✔️] v201-forgeendertech-1.20.1-11.1.0.1-build.0289.jar
  [✔️] v201-formations-1.0.2-forge-mc1.20.jar
  [✔️] v201-formationsnether-1.0.2.jar
  [✔️] v201-formationsoverworld-1.0.1.jar
  [✔️] v201-framedcompactdrawers-1.20-6.0.0.jar
  [✔️] v201-framework-forge-1.20.1-0.6.16.jar
  [✔️] v201-fullbrightnesstoggle-1.20.1-3.2.jar
  [✔️] v201-gamemenumodoption-mc1.20.1-2.2.1.jar
  [✔️] v201-geophilic-v2.1.0-mc1.20u1.20.2.jar
  [✔️] v201-gildednetherite-cj-forge-mc1.20-1.0.2.jar
  [✔️] v201-global-packs-forge-1.19.4-1.16.2-forge.jar
  [✔️] v201-goblintraders-forge-1.20.1-1.9.3.jar
  [✔️] v201-guardvillagers-1.20.1-1.6.3.jar
  [✔️] v201-hopobettermineshaft-1.20-1.20.1-1.1.8.jar
  [✔️] v201-hopobetterruinedportals-1.20-1.20.2-1.3.7.jar
  [✔️] v201-hopobetterunderwaterruins-1.20-1.20.2-1.1.4b.jar
  [✔️] v201-immediatelyfast-forge-1.2.8-1.20.4.jar
  [✔️] v201-imst-2.1.0.jar
  [✔️] v201-incendium-1.20.4-v5.3.4.jar
  [✔️] v201-inventoryprofilesnext-forge-1.20-1.10.9.jar
  [✔️] v201-invhud.forge.1.20.1-3.4.18.jar
  [✔️] v201-ironchest-1.20.1-14.4.4.jar
  [✔️] v201-ironshulkerbox-1.20.1-5.3.2.jar
  [✔️] v201-itemcollectors-1.1.8-forge-mc1.20.jar
  [✔️] v201-jade-1.20.1-forge-11.6.4.jar
  [✔️] v201-jadeaddons-1.20.1-forge-5.2.2.jar
  [✔️] v201-jearchaeology-1.20.1-1.0.3.jar
  [✔️] v201-jeed-1.20-2.1.6.jar
  [✔️] v201-jei-1.20.1-forge-15.2.0.27.jar
  [✔️] v201-jeitweaker-forge-1.20.1-8.0.6.jar
  [✔️] v201-jess-1.20.1-forge-1.1.0.jar
  [✔️] v201-journeymap-1.20.1-5.9.18-forge.jar
  [✔️] v201-justenoughadvancements-1.20.1-5.0.1.jar
  [✔️] v201-justenoughbeacons-forge-1.19-1.1.1.jar
  [✔️] v201-justenoughbreeding-forge-1.20.x-1.0.12.jar
  [✔️] v201-justenoughmekanismmultiblocks-1.20.1-4.2.jar
  [✔️] v201-justenoughprofessions-forge-1.20.1-3.0.1.jar
  [✔️] v201-justenoughresources-1.20.1-1.4.0.238.jar
  [✔️] v201-kambrik-6.1.1-1.20.1-forge.jar
  [✔️] v201-kotlinforforge-4.9.0-all.jar
  [✔️] v201-leavesbegone-v8.0.0-1.20.1-forge.jar
  [✔️] v201-libipn-forge-1.20-4.0.1.jar
  [✔️] v201-libraryferret-forge-1.20.1-4.0.0.jar
  [✔️] v201-libx-1.20.1-5.0.7.jar
  [✔️] v201-light-overlay-8.0.0-forge.jar
  [✔️] v201-lootbeams-1.20.1-1.2.3.jar
  [✔️] v201-mcjtylib-1.20-8.0.3.jar
  [✔️] v201-mcw-bridges-2.1.1-mc1.20.1forge.jar
  [✔️] v201-mcw-doors-1.1.0forge-mc1.20.1.jar
  [✔️] v201-mcw-fences-1.0.7-mc1.20.1forge.jar
  [✔️] v201-mcw-furniture-3.2.1-mc1.20.1forge.jar
  [✔️] v201-mcw-lights-1.0.6-mc1.20.1forge.jar
  [✔️] v201-mcw-roofs-2.2.4b-mc1.20.1forge.jar
  [✔️] v201-mcw-trapdoors-1.1.2-mc1.20.1forge.jar
  [✔️] v201-mcw-windows-2.2.1-mc1.20.1forge.jar
  [✔️] v201-mekanism-1.20.1-10.4.5.19.jar
  [✔️] v201-mekanismadditions-1.20.1-10.4.5.19.jar
  [✔️] v201-mekanismgenerators-1.20.1-10.4.5.19.jar
  [✔️] v201-mekanismtools-1.20.1-10.4.5.19.jar
  [✔️] v201-memoryleakfix-forge-1.17-1.1.2.jar
  [✔️] v201-modelfix-1.14.jar
  [✔️] v201-modernfix-forge-5.11.1-mc1.20.1.jar
  [✔️] v201-modernui-forge-1.20.1-3.9.0.4-universal.jar
  [✔️] v201-moonlight-1.20-2.8.67-forge.jar
  [✔️] v201-morejeiinfo-forge-1.20.1-1.0.1.jar
  [✔️] v201-mousetweaks-forge-mc1.20-2.25.jar
  [✔️] v201-mvs-4.1.1-1.20-forge.jar
  [✔️] v201-myserveriscompatible-1.20-1.0.jar
  [✔️] v201-neat-1.20-35-forge.jar
  [✔️] v201-netheritehorsearmor-cj-forge-mc1.20-1.4.0.jar
  [✔️] v201-netherportalfix-forge-1.20-13.0.1.jar
  [✔️] v201-netherportalspread-1.20.1-7.8.jar
  [✔️] v201-nofog-1.3.4-1.20.1-forge.jar
  [✔️] v201-nomowanderer-1.20.1-1.6.2.jar
  [✔️] v201-nonconflictkeys-forge-1.19.x-1.20-1.0.0.jar
  [✔️] v201-no-telemetry-1.8.0.jar
  [✔️] v201-notenoughwands-1.20-6.0.1.jar
  [✔️] v201-oculus-mc1.20.1-1.6.13.jar
  [✔️] v201-overflowingbars-v8.0.0-1.20.1-forge.jar
  [✔️] v201-patchouli-1.20.1-81-forge.jar
  [✔️] v201-philips-ruins1.20.1-2.8.jar
  [✔️] v201-placebo-1.20.1-8.3.7.jar
  [✔️] v201-polymorph-forge-0.49.2-1.20.1.jar
  [✔️] v201-puzzleslib-v8.1.11-1.20.1-forge.jar
  [✔️] v201-repurposed-structures-7.1.11-1.20.1-forge.jar
  [✔️] v201-resourcepackoverrides-v8.0.2-1.20.1-forge.jar
  [✔️] v201-restrictions-1.20-6.0.1.jar
  [✔️] v201-rex-s-additionalstructures-1.20.x-v.4.2.1.jar
  [✔️] v201-rftoolsbase-1.20-5.0.2.jar
  [✔️] v201-rftoolsbuilder-1.20-6.0.1.jar
  [✔️] v201-rftoolscontrol-1.20-7.0.1.jar
  [✔️] v201-rftoolsdim-1.20-11.0.5.jar
  [✔️] v201-rftoolspower-1.20-6.0.1.jar
  [✔️] v201-rftoolsstorage-1.20-5.0.1.jar
  [✔️] v201-rftoolsutility-1.20-6.0.3.jar
  [✔️] v201-searchables-forge-1.20.1-1.0.2.jar
  [✔️] v201-shadowizardlib-1.20.1-1.0.7.jar
  [✔️] v201-shrines-1.20.1-6.0.2.jar
  [✔️] v201-signtastic-1.20-3.0.0.jar
  [✔️] v201-simplestoragenetwork-1.20.1-1.10.0.jar
  [✔️] v201-simplyhouses-1.1.4-1.20.1-forge.jar
  [✔️] v201-smoothboot-mc1.20.1-0.0.3.jar
  [✔️] v201-smoothchunk-1.20.1-3.5.jar
  [✔️] v201-spark-1.10.53-forge.jar
  [✔️] v201-spawnermod-1.20.1-1.9.3-forge.jar
  [✔️] v201-storagedrawers-1.20.1-12.0.2.jar
  [✔️] v201-structory-1.20.2-v1.3.4.jar
  [✔️] v201-structureessentials-1.20.1-3.2.jar
  [✔️] v201-subterrestrial-forge-1.20-2.2.0.jar
  [✔️] v201-supermartijn642configlib-1.1.8-forge-mc1.20.jar
  [✔️] v201-supermartijn642corelib-1.1.16-forge-mc1.20.1.jar
  [✔️] v201-supplementaries-1.20-2.6.12.jar
  [✔️] v201-tagtooltips-1.20.1-1.0.jar
  [✔️] v201-terrablender-forge-1.20.1-3.0.0.169.jar
  [✔️] v201-tony-s-structures-v3.jar
  [✔️] v201-towns-and-towers-1.12-fabric-forge.jar
  [✔️] v201-trimeffects-1.0.3-forge.jar
  [✔️] v201-undergroundvillages-forge-1.20.1-2.1.0.jar
  [✔️] v201-universalenchants-v8.0.0-1.20.1-forge.jar
  [✔️] v201-villagersplus-3.0-forge-for-1.20.1.jar
  [✔️] v201-waystones-forge-1.20-14.1.3.jar
  [✔️] v201-xnet-1.20-6.0.2.jar
  [✔️] v201-yeetusexperimentus-forge-2.3.1-build.6-mc1.20.1.jar
  [✔️] v201-yungsapi-1.20-forge-4.0.2.jar
  [✔️] v201-yungsbetterdeserttemples-1.20-forge-3.0.3.jar
  [✔️] v201-yungsbetterdungeons-1.20-forge-4.0.3.jar
  [✔️] v201-yungsbetterendisland-1.20-forge-2.0.4.jar
  [✔️] v201-yungsbetterjungletemples-1.20-forge-2.0.4.jar
  [✔️] v201-yungsbettermineshafts-1.20-forge-4.0.4.jar
  [✔️] v201-yungsbetternetherfortresses-1.20-forge-2.0.5.jar
  [✔️] v201-yungsbetteroceanmonuments-1.20-forge-3.0.4.jar
  [✔️] v201-yungsbetterstrongholds-1.20-forge-4.0.3.jar
  [✔️] v201-yungsbetterwitchhuts-1.20-forge-3.0.3.jar
  [✔️] v201-yungsbridges-1.20-forge-4.0.3.jar
  [✔️] v201-yungsextras-1.20-forge-4.0.3.jar

**Comment by esvarc** [comment]
Here are java Arguments for MultiMC 5 launcher and memory settings were 2G minimum and 9G max, but max could be 4G (top mem alloc was around 3GB).

-XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=200 -XX:G1HeapRegionSize=32M -XX:InitiatingHeapOccupancyPercent=70 -XX:ParallelGCThreads=20 -XX:ConcGCThreads=5

**Comment by J-Dill** [comment]
@Zelioth Thanks for the report, and I can see that you mentioned using NoMoWanderer v1.3.12: try updating to v1.3.13 as this version should fix the issue you are seeing.

**Comment by J-Dill** [comment]
@esvarc I will try recreating what you are seeing here... Will let you know what I find. That version of NoMoWanderer you are using should have this issue fixed so I will take a look.

**Comment by J-Dill** [comment]
@esvarc In your mod list, it says that you are using `nomowanderer-1.20.1-1.6.2.jar`, but in the error it mentions:
```
...TRANSFORMER/nomowanderer@1.20.1_1.6.0/nomowanderer.world.EntitySpawnHandler...
```
Which leads me to think you may actually be using NoMoWanderer v1.6.0. Can you double-check which version of NoMoWanderer is installed on your client/server? If you are using v1.6.2 then feel free to make a new issue, but I will be closing this issue as I currently see no signs of a mod issue.

**Comment by esvarc** [comment]
I don't know how it is possible that version 1.6.0 is reported in the log. The JAR file is downloaded from CurseForge. How can you be convinced from the package I put on OneDrive, where a complete instance of all mods is packed and a test save of the game is saved. You can see from the package that there is a file v201-nomowanderer-1.20.1-1.6.2.jar, the prefix v201- is a part that I add to the file name to distinguish the version, but otherwise it is a file according to CurseForge. Since the game is started in single player, of course the same version is on the server.

I downloaded the current version 1.6.3. from Curse forge, put it in the given instance packed also on my OneDrive in the link https://1drv.ms/u/s!AsuclbmnGMOphrNnYmqiRs6Egrjb6w?e=YbSPZ7, and again after teleporting to the problematic chunk the game crashed.

`/teleport @a -640 ~ 144`

In the log, a similar message in the given function

```
"Server thread" prio=5 Id=142 TIMED_WAITING on java.lang.String@4de3cf53
	at java.base@17.0.8.1/jdk.internal.misc.Unsafe.park(Native Method)
	-  waiting on java.lang.String@4de3cf53
	at java.base@17.0.8.1/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_5667_(BlockableEventLoop.java:151)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:141)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerChunkCache.getChunkBlocking(ServerChunkCache.java:2768)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerChunkCache.m_7587_(ServerChunkCache.java:2664)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.Level.m_6522_(Level.java:187)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.Level.m_46865_(Level.java:1543)
	at TRANSFORMER/nomowanderer@1.20.1_1.6.3/nomowanderer.world.EntitySpawnHandler.getAABB(EntitySpawnHandler.java:132)
	at TRANSFORMER/nomowanderer@1.20.1_1.6.3/nomowanderer.world.EntitySpawnHandler.canFindCancelEntity(EntitySpawnHandler.java:91)
	at TRANSFORMER/nomowanderer@1.20.1_1.6.3/nomowanderer.world.EntitySpawnHandler.checkSpawn(EntitySpawnHandler.java:66)
	at TRANSFORMER/nomowanderer@1.20.1_1.6.3/nomowanderer.world.EntitySpawnHandler.maybeChangeEntitySpawn(EntitySpawnHandler.java:40)
	at TRANSFORMER/nomowanderer@1.20.1_1.6.3/nomowanderer.world.__EntitySpawnHandler_maybeChangeEntitySpawn_EntityJoinLevelEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus$$Lambda$5058/0x0000025eaedf6c48.invoke(Unknown Source)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:315)
	at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:296)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157538_(PersistentEntitySectionManager.java:79)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157604_(PersistentEntitySectionManager.java:121)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.PersistentEntitySectionManager$$Lambda$33563/0x0000025eb12c8828.accept(Unknown Source)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType$1.m_147060_(EntityType.java:585)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType$1$$Lambda$31707/0x0000025eb0fe3108.apply(Unknown Source)
	at java.base@17.0.8.1/java.util.Optional.map(Optional.java:260)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType.m_20645_(EntityType.java:563)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType$1.m_147056_(EntityType.java:584)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType$1$$Lambda$31706/0x0000025eb0fe2ed8.accept(Unknown Source)
	at java.base@17.0.8.1/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.EntityType$1.tryAdvance(EntityType.java:583)
	at java.base@17.0.8.1/java.util.Spliterator.forEachRemaining(Spliterator.java:332)
	at java.base@17.0.8.1/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157559_(PersistentEntitySectionManager.java:120)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_143327_(ServerLevel.java:1434)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.m_143064_(ChunkMap.java:690)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.m_214898_(ChunkMap.java:707)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap$$Lambda$32395/0x0000025eb1109358.m_196866_(Unknown Source)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.chunk.LevelChunk.m_62952_(LevelChunk.java:438)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.mixinextras$bridge$m_62952_$142(ChunkMap.java)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap$$Lambda$31678/0x0000025eb0fdf7f8.call(Unknown Source)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.wrapOperation$zdn000$setCurrentLoadingThenPostLoad(ChunkMap.java:1905)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.m_214854_(ChunkMap.java:715)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap$$Lambda$31674/0x0000025eb0fdf398.apply(Unknown Source)
	at MC-BOOTSTRAP/datafixerupper@6.0.8/com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162)
	at MC-BOOTSTRAP/datafixerupper@6.0.8/com.mojang.datafixers.util.Either$$Lambda$31675/0x0000025eb0fd1fe0.apply(Unknown Source)
	at MC-BOOTSTRAP/datafixerupper@6.0.8/com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at MC-BOOTSTRAP/datafixerupper@6.0.8/com.mojang.datafixers.util.Either.mapLeft(Either.java:162)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap.m_287044_(ChunkMap.java:699)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkMap$$Lambda$31671/0x0000025eb0fded18.apply(Unknown Source)
	at java.base@17.0.8.1/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base@17.0.8.1/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkTaskPriorityQueueSorter.m_143188_(ChunkTaskPriorityQueueSorter.java:62)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ChunkTaskPriorityQueueSorter$$Lambda$31620/0x0000025eb0fc7000.run(Unknown Source)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_6367_(BlockableEventLoop.java:156)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_6367_(ServerChunkCache.java:541)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_7245_(BlockableEventLoop.java:130)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerChunkCache$MainThreadExecutor.m_7245_(ServerChunkCache.java:549)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerChunkCache.m_8466_(ServerChunkCache.java:269)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_129961_(MinecraftServer.java:758)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_7245_(MinecraftServer.java:747)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.util.thread.BlockableEventLoop.m_18701_(BlockableEventLoop.java:139)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130012_(MinecraftServer.java:733)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:665)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251)
	at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer$$Lambda$31226/0x0000025eb0f2abf0.run(Unknown Source)
	at java.base@17.0.8.1/java.lang.Thread.run(Thread.java:833)
```

**Comment by esvarc** [comment]
Sorry for the late response, but the GitHub notification ended up in spam and I found it while cleaning out the spam folder.

However, the instructions on how to reproduce the error are valid, the entire installation is available packaged as a ZIP, just replace v201-nomowanderer-1.20.1-1.6.2.jar with nomowanderer-1.20.1_1.6.3.jar. In my test I did this and downloaded the current version from https://www.curseforge.com/minecraft/mc-mods/nomowanderer.

In my case I'm using Forge version 47.2.20 and MC version 1.20.1
